### PR TITLE
doc: add example for building with constrained resources

### PR DESCRIPTION
**Pull request by jarret**
discussed in github issue #6658

### CODE REVIEW

File: doc/build-unix.md [hunk 1]
```diff
@@ -54,8 +54,11 @@ System requirements
 --------------------
 
 C++ compilers are memory-hungry. It is recommended to have at least 1 GB of
-memory available when compiling Bitcoin Core. With 512MB of memory or less
-compilation will take much longer due to swap thrashing.
+memory available when compiling Bitcoin Core. However, if you are building on a
+system with less than 1 GB memory, the compiler can be instructed to conserve
+memory with additional CXXFLAGS:
+
+    ./configure CXXFLAGS="--param ggc-min-expand=1 --param ggc-min-heapsize=32768"
```
Comment (jonasschnelli) [inline doc/build-unix.md]: nit. Looks like gcc specific. Maybe okay. But is there a clang equivalent or something that is supported by "all"/"most" compilers?

**Comment by jonasschnelli** [comment]
utACK

File: doc/build-unix.md (hunk 1 above)
Comment (laanwj) [inline doc/build-unix.md]: Nit: From what I understood of #6658, these settings may even be helpful with only 1 GB of memory, not just with less

Comment (jarret) [inline doc/build-unix.md]: @jonasschnelli:

I have updated the wording to indicate gcc-specificness. I will have to do more
research and testing for a working low-memory clang recipe. With gcc and
aggressive values for these params, I have been able to successfully build with
on a Raspberry Pi 1 with 512 RAM and no swap partition. This is better than
@laanwj's clang measurements posted on issue 6658, so until we can demonstrate
a better tweak with clang, this appears to be a solid recommendation.


@laanwj:

That is correct, thanks. I have also updated the commit with a change of the
section title and a phrasing that feels better to me. I went with a wording to
encourage people to try if they are below 1.5 GB. Please let me know if you
think it is an improvement.

**Comment by laanwj** [comment]
utACK

**Pull request merged**